### PR TITLE
Code Review - Escaping, reference operators, url protocols

### PR DESCRIPTION
**Pull request by christianc1**
Original review was done for @campusinsiders on @wpcomvip by @tomjn.

Figuring the changeset is valuable for others as well, I added some additional missed escaping and put it all together as a pull request.

.org users:
tjnowell (Tom)
christian1012 (me)

### CODE REVIEW

File: common/lib/acm-provider.php [hunk 1]
```diff
@@ -63,16 +65,17 @@ public function action_do_robotstxt() {
...
-		echo 'User-agent: ' . $this->crawler_user_agent . PHP_EOL;
+		echo esc_html( 'User-agent: ' . $this->crawler_user_agent . PHP_EOL );
 
 		foreach ( $disallowed as $disallow ) {
-			echo 'Disallow: ' . $disallow . PHP_EOL;
+			echo esc_html( 'Disallow: ' . $disallow . PHP_EOL );
 		}
 
-		echo PHP_EOL;
+		echo esc_html( PHP_EOL );
```
Comment (joshbetz) [inline common/lib/acm-provider.php]: Constants and string literals don't need to be escaped.

Comment (christianc1) [inline common/lib/acm-provider.php]: updated in 60c50cfff4c6bcbe72759d28a2215e7752f3149c

**Comment by christianc1** [comment]
@joshbetz : Following up on this, anything additional I can do to help this along?

File: common/views/ad-code-manager.tpl.php [hunk 2]
```diff
@@ -114,7 +114,7 @@
 ?>
 <div class="form-field form-required">
 	<label for="<?php echo esc_attr( $column_id ) ?>"><?php echo esc_html( $arg['label'] ) ?></label>
-	<input name="<?php echo esc_attr( $column_id ) ?>" id="<?php echo esc_attr( $column_id ) ?>" type="text" value="" size="40" aria-required="<?php echo $arg['required'] ?>">
+	<input name="<?php echo esc_attr( $column_id ) ?>" id="<?php echo esc_attr( $column_id ) ?>" type="text" value="" size="40" aria-required="<?php echo esc_url( $arg['required'] ); ?>">
```
Comment (GaryJones) [inline common/views/ad-code-manager.tpl.php:117]: This should be `esc_attr()`.

**Review by GaryJones — COMMENTED** [review]
I've cherry-picked some of these commits into a new branch, and fixed up the issues I've highlighted.

**Comment by GaryJones** [comment]
Closing this PR, as the new branch will contain some of the work done in this PR. Thank you for your contribution!

**Pull request closed without merging**